### PR TITLE
Fix rounding issue when sending max tokens

### LINKED ISSUE #5682: Issue with sending max tokens/tokens amount that has long decimals.
When viewing this token transfer tx https://etherscan.io/tx/0x5b3765cdf02648a42e0f307aa0b0d4a72d4151ecd13fd993f1decf5b2200ac35#decodetab, under the Input Data section, the value is 57888134750431890000. After reviewing  token value of the from address, https://etherscan.io/token/0xe41d2489571d322189246dafa5ebde1f4699f498?a=0x10ac56c7adac67d1a50002540c7a000bb5b12530, seems we are rounding up at the end.

TLDR; 
Input Value:    57888134750431890000
Actual Value: 57888134750431889787

### PR DESCRIPTION
**Pull request by alextsg**
Fixes #5682

### CODE REVIEW

File: test/e2e/beta/metamask-beta-ui.spec.js [hunk 1]
```diff
@@ -894,8 +894,8 @@ describe('MetaMask', function () {
       // test cancelled on firefox until https://github.com/mozilla/geckodriver/issues/906 is resolved,
       // or possibly until we use latest version of firefox in the tests
       if (process.env.SELENIUM_BROWSER !== 'firefox') {
-        const tokenBalanceAmount = await findElement(driver, By.css('.transaction-view-balance__token-balance'))
-        assert.equal(await tokenBalanceAmount.getText(), '43 TST')
+        const tokenBalanceAmount = await findElement(driver, By.css('.transaction-view-balance__primary-balance'))
+        assert.equal(await tokenBalanceAmount.getText(), '43\nTST')
```
Comment (whymarrh) [inline test/e2e/beta/metamask-beta-ui.spec.js]: We could use `\s` here in a regular expression, similar to how we have below:

```js
await driver.wait(until.elementTextMatches(balance, /0\s*BAT/))
```

File: ui/app/components/currency-display/currency-display.container.js [hunk 2]
```diff
@@ -20,15 +20,28 @@ const mergeProps = (stateProps, dispatchProps, ownProps) => {
...
+  if (typeof displayValue === 'undefined') {
+    const convertedValue = getValueFromWeiHex({
+      value,
+      fromCurrency: nativeCurrency,
+      toCurrency, conversionRate,
+      numberOfDecimals,
+      toDenomination: denomination,
+    })
+
+    displayValue = propsDisplayValue || formatCurrency(convertedValue, toCurrency)
```
Comment (whymarrh) [inline ui/app/components/currency-display/currency-display.container.js]: If we inline `convertedValue` could we write this as a single `cost displayValue` statement without the conditional?

**Review by whymarrh — CHANGES_REQUESTED** [review]
Two small comments but otherwise this looks good to me

File: ui/app/components/send/send.component.js [hunk 3]
```diff
@@ -11,6 +11,9 @@ import {
 import SendHeader from './send-header/'
 import SendContent from './send-content/'
 import SendFooter from './send-footer/'
+import BigNumber from 'bignumber.js'
+
+window.BigNumber = BigNumber
```
Comment (danjm) [inline ui/app/components/send/send.component.js]: why do we need to put `BigNumber` on `window`? I'm guess this was just for local development/testing purposes?

**Review by whymarrh — DISMISSED** [review]

**Comment by danjm** [comment]
I made a PR against this branch to fix a minor issue: https://github.com/alextsg/metamask-extension/pull/2

**Comment by danjm** [comment]
QA'd. All looks good.

As a separate issue, we may want to deal with cases with a lot of significant digits on the confirm screen

screenshot from 2018-11-11 19-45-45

**Comment by alextsg** [comment]
@danjm Merged your PR and noticed that tests were failing due to number inputs. Why was it necessary to remove the `Number()` part from `return Number(decimalValueString) || 0` in `token-input.component.js`? `getDecimalValue` is expected to return a number instead of a string so it's causing some input issues.

**Comment by danfinlay** [comment]
Fixes #5180.

**Review by danjm — APPROVED** [review]

**Pull request merged**